extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/scan/ScanQueryFrameProcessor.java

```diff
@@ -198,6 +198,9 @@ private static ScanQuery prepareScanQueryForDataServer(@NotNull ScanQuery scanQu
   @Override
   protected ReturnOrAwait<SegmentsInputSlice> runWithDataServerQuery(final DataServerQueryHandler dataServerQueryHandler) throws IOException
   {
+    // Try to run with the segment on the server
+    // if segment was not found for some reason
```

Comment:
This comments are not accurate. I would suggest removing them.

Reply:
If segment is handed off, only then fetch from deep storage is the correct flow.

Reply:
Addressed